src/components/controller/loader.ts

```diff
@@ -0,0 +1,49 @@
+import { IDataSources } from "../view/sources/sources";
+
+export type Callback<T> = (data: T) => void;
+class Loader {
+    baseLink: string;
+    readonly options: { [key: string]: string };
+
+    constructor(baseLink: string, options: { [key: string]: string }) {
+        this.baseLink = baseLink;
+        this.options = options;
+    }
+
+    public getResp(
+        { endpoint, options = {} }: {endpoint: string, options?: object},
+        callback: Callback<IDataSources>) {
+        this.load('GET', endpoint, callback, options);
+    }
+
+    errorHandler(res: Response) {
+        if (!res.ok) {
+            if (res.status === 401 || res.status === 404)
```

Comment:
Please use constants or Enum for status codes.

Reply:
Add enum for error status codes.